### ISSUE TITLE
Request: author and title for epub files on the homescreen

### ISSUE THREAD

**Issue by slow-mo** [koreader/kpvbooklet#16]
Would it be hard to make kpvbooklet show the author and title for epub files on the homescreen - instead of the filename - as for pdf and mobi/azw files?

**Comment by chrox** [comment]
It's possible but needs a lot of reverse engineering to implement an EPUB metadata extractor like the PDF metadata extractor provided by Kindle system.

**Comment by dracodoc** [comment]
Isn't EPUB metadata an open standard? I think any epub is just a zip, and you can find metadata in content.opf (the location of this file may be different) like this:

        <meta name="calibre:title_sort" content="Synaptic Self"/>
        <dc:language>zh</dc:language>
        <dc:creator opf:file-as="LeDoux, Joseph" opf:role="aut">Joseph LeDoux</dc:creator>

You can also find the book cover, which is another good option to show in the book cover view screen.

     <manifest>
        <item href="cover.jpeg" id="cover" media-type="image/jpeg"/>

**Comment by chrox** [comment]
It's easy to extract the meta data of a EPUB document indeed. But it's harder to integrate with the Kindle native system. Because Amazon never provides developing SDK for the Kindle system. Projects like the `kpvbooklet` have to reverse engineering the interface and some implementations of the Kindle system in order to inter-operate the digital content in Kindle devices.

**Comment by dracodoc** [comment]
Yes that is a problem. I think calibre did some reverse engineering so that book transferred by calibre can show cover image, while just copy books to kindle will never have the cover shown. Maybe calibre's implementation can be a reference?

As for showing book author and title, there is a simple but ugly hack without need of reverse engineering:

_**just extract meta data from book then rename the file to `title - by author.epub`.**_ Then file name get shown, which actually have the book title and author.

Calibre actually always rename books to its own way by metadata.

**Comment by NiLuJe** [comment]
Calibre doesn't have to directly interact with the Kindle Framework.
KPVBooklet does. It's far, far, far more complex than that, and the hoops Calibre has to jump through to make the framework happy are barely related ;).

**Comment by martinrotter** [comment]
So I am also bothered by this issue. If kpvbooklet cannot display the actual metadata, I would be satisfied with this:

- Does kpvbooklet support displaying of non-English characters in filenames of books?
- If so, is there a way to force Calibre to actually "send to device" book files with non-English characters included?

I mean, I searched throught Google like a crazy, but could not find a solution. I have big and nicely organized library and each book has filled title/description/author in Czech language. For example: Souostroví Gulag/Alexandr Solženicyn (in translation Gulag Archipellago, Solzhenicyn). But when I "send to device" the file does not contain non-English characters and those are replaced by "ASCII" ones. Is there a way to force Calibre to send correct filenames for my whole library?

**Comment by dracodoc** [comment]
The calibre question is unrelated, you can disable preference - save to disk - convert non-english characters, and set saving template to what you want, then use saving to disk instead of send to device to transfer books (save to your kindle partition).

I'm not using this, so I'm not sure if this have any difference from send to device.

**Comment by cytown** [comment]
If want to implement this, maybe you need implement specified com.amazon.ebook.booklet.reader.impl.detail.ReaderDetailViewFactory


Right?

**Comment by diegombeltran** [comment]
Can it be implemented?
